kythe/java/com/google/devtools/kythe/extractors/java/JavaCompilationUnitExtractor.java

```diff
@@ -252,6 +252,7 @@ public CompilationDescription extract(
       Iterable<String> sourcepath,
       Iterable<String> processorpath,
       Iterable<String> processors,
+      Optional<String> systemDir,
```

Comment:
I'd really, really like to avoid the proliferation of arguments in this already-too-long argument list.

Can you just make this a public member function on the JavaCompilationUnit extractor instead?

`useSystemDirectory()` or similar?

Reply:
done